examples/gallery/lines/line_segment_ends.py

```diff
@@ -0,0 +1,75 @@
+"""
+Line segment caps and joints
+============================
+PyGMT offers different appearances of line segment caps and joints. The desired
+appearance can be set via the GMT default parameters :gmt-term:`PS_LINE_CAP`
+(``"butt"``, ``"round"``, or ``"square"`` [Default]) as well as :gmt-term:`PS_LINE_JOIN`
+(``"bevel"``, ``"round"``, and ``"miter"`` [Default]) and :gmt-term:`PS_MITER_LIMIT`
+(limit on the angle at the mitered joint below which a bevel is applied).
+"""
+
+# %%
+import numpy as np
+import pygmt
+
+# Set up dictionary for colors
+dict_col = {
+    "round": "green4",
+    "square": "steelblue",
+    "butt": "orange",
+    "miter": "steelblue",
+    "bevel": "orange",
+}
+
+# Create new Figure instance
+fig = pygmt.Figure()
+
+# -----------------------------------------------------------------------------
+# Top: PS_LINE_CAP
+
+# Create sample data
+x = np.array([30, 170])
+y = np.array([70, 70])
+
+fig.basemap(region=[0, 260, 0, 100], projection="x1p", frame="rltb")
+
+for line_cap in ["butt", "round", "square"]:
+    # Change GMT default locally using a context manager
```

Comment:
```suggestion
    # Change GMT default locally
```
Should be clear without an extended description what the with statement does.